device/cluster.cpp

```diff
@@ -809,77 +992,87 @@ void Cluster::broadcast_pcie_tensix_risc_reset(chip_id_t chip_id, const TensixSo
-               std::find(get_soc_descriptor(target_device).ethernet_cores.begin(), get_soc_descriptor(target_device).ethernet_cores.end(), core) != get_soc_descriptor(target_device).ethernet_cores.end(),
-                                "Cannot deassert reset on a non-tensix or harvested core");
-    bool target_is_mmio_capable = ndesc -> is_chip_mmio_capable(target_device);
-    if(target_is_mmio_capable) {
-        log_assert(m_pci_device_map.find(target_device) != m_pci_device_map.end(), "Could not find MMIO mapped device in devices connected over PCIe");
+void Cluster::deassert_risc_reset_at_core(tt_cxy_pair core, const TensixSoftResetOptions& soft_resets) {
+    std::uint32_t target_device =
+        core.chip;  // Get Target Device to query soc descriptor and determine location in cluster
+    log_assert(
+        std::find(
+            get_soc_descriptor(target_device).workers.begin(), get_soc_descriptor(target_device).workers.end(), core) !=
+                get_soc_descriptor(target_device).workers.end() ||
+            std::find(
+                get_soc_descriptor(target_device).ethernet_cores.begin(),
+                get_soc_descriptor(target_device).ethernet_cores.end(),
+                core) != get_soc_descriptor(target_device).ethernet_cores.end(),
+        "Cannot deassert reset on a non-tensix or harvested core");
+    bool target_is_mmio_capable = ndesc->is_chip_mmio_capable(target_device);
+    if (target_is_mmio_capable) {
+        log_assert(
+            m_pci_device_map.find(target_device) != m_pci_device_map.end(),
+            "Could not find MMIO mapped device in devices connected over PCIe");
         send_tensix_risc_reset_to_core(core, soft_resets);
-    }
-    else {
+    } else {
         log_assert(arch_name != tt::ARCH::BLACKHOLE, "Can't issue access to remote core in BH");
         send_remote_tensix_risc_reset_to_core(core, soft_resets);
     }
 }
 
 void Cluster::assert_risc_reset_at_core(tt_cxy_pair core) {
-    std::uint32_t target_device = core.chip; // Get Target Device to query soc descriptor and determine location in cluster
-    log_assert(std::find(get_soc_descriptor(target_device).workers.begin(), get_soc_descriptor(target_device).workers.end(), core) != get_soc_descriptor(target_device).workers.end() ||
-               std::find(get_soc_descriptor(target_device).ethernet_cores.begin(), get_soc_descriptor(target_device).ethernet_cores.end(), core) != get_soc_descriptor(target_device).ethernet_cores.end(),
-                                "Cannot assert reset on a non-tensix or harvested core");
-    bool target_is_mmio_capable = ndesc -> is_chip_mmio_capable(target_device);
-    if(target_is_mmio_capable) {
-        log_assert(m_pci_device_map.find(target_device) != m_pci_device_map.end(), "Could not find MMIO mapped device in devices connected over PCIe");
+    std::uint32_t target_device =
+        core.chip;  // Get Target Device to query soc descriptor and determine location in cluster
```

Comment:
again, comment above

Reply:
This was because of comment in continuation of the line. Line break was removed when I moved the comment.